server/src/database/migrations/001-initial-migration.ts

```diff
@@ -0,0 +1,384 @@
+// SPDX-FileCopyrightText: 2022 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import { QueryInterface, DataTypes, Transaction } from 'sequelize';
+
+export default {
+  up: async (queryInterface: QueryInterface): Promise<void> => {
+    const transaction: Transaction = await queryInterface.sequelize.transaction();
+    try {
+      await queryInterface.createTable('user', {
+        id: {
+          type: DataTypes.INTEGER,
+          autoIncrement: true,
+          primaryKey: true
+        },
+        student_id: {
+          type: new DataTypes.STRING,
+          unique: true,
+          allowNull: false
+        },
+        name: {
+          type: new DataTypes.STRING,
+          allowNull: false
+        },
+        email: {
+          type: new DataTypes.STRING(255),
+          allowNull: false,
+          unique: true,
+          validate: {
+            isEmail: true
+          }
+        },
+        password: {
+          type: new DataTypes.CHAR(60),
+          allowNull: false
+        },
+        created_at: DataTypes.DATE,
+        updated_at: DataTypes.DATE
+      }, { transaction });
+      await queryInterface.createTable('course', {
+        id: {
+          type: DataTypes.INTEGER,
+          autoIncrement: true,
+          primaryKey: true
+        },
+        course_code: {
+          type: new DataTypes.STRING,
+          allowNull: false
+        },
+        min_credits: {
+          type: new DataTypes.INTEGER,
+          allowNull: false
+        },
+        max_credits: {
+          type: new DataTypes.INTEGER,
+          allowNull: false,
+        },
+        created_at: DataTypes.DATE,
+        updated_at: DataTypes.DATE
+      }, { transaction });
+      await queryInterface.createTable('course_instance', {
+        id: {
+          type: DataTypes.INTEGER,
+          autoIncrement: true,
+          primaryKey: true
+        },
+        course_id: {
+          type: DataTypes.INTEGER,
+          allowNull: false,
+          references: {
+            model: 'course',
+            key: 'id'
+          },
+          onDelete: 'CASCADE',
+          onUpdate: 'CASCADE'
+        },
+        grading_type: {
+          type: DataTypes.ENUM('PASSFAIL', 'NUMERICAL'),
+          allowNull: false
+        },
+        teaching_period: {
+          type: DataTypes.STRING,
```

Comment:
Could be if we want. We would just have to map out all the possible period combinations, e.g., 'I', 'II', 'I-II',....

Reply:
Right, I completely forgot about the possibility of a course spanning multiple periods. In that case keeping it this way may be simpler, or possibly this could be split to `starting_period` and `ending_period` with both having enum values.

Reply:
separating start and end could be better option, makes it easier to implement for example search functionality "find courses starting on period X".

Reply:
implemented [ae2e4ba9df8f3c319cdf1a629292c8634bbaaf2c](https://github.com/aalto-grades/base-repository/pull/51/commits/ae2e4ba9df8f3c319cdf1a629292c8634bbaaf2c)